### ISSUE TITLE
RPC calls return null for blocks between 19500000 and 19599999

### ISSUE THREAD

**Issue by jack695** [ledgerwatch/erigon#10198]
#### System information

Erigon version: `2.59.1`

OS & Version: Ubuntu 22.04.3 LTS

Note: Erigon is running inside Docker: thorax/erigon.

Commit hash: 02c9d4ba

Erigon Command (with flags/config):

erigon
  --authrpc.addr=0.0.0.0
  --authrpc.vhosts=any
  --authrpc.jwtsecret=/secrets/jwt.hex
  --http.addr=0.0.0.0
  --http.vhosts=any
  --datadir=/data
  --http.api=eth,erigon,engine,net,debug,trace

Consensus Layer: Lighthouse

Consensus Layer Command (with flags/config):
lighthouse
      bn
      --network=mainnet
      --execution-endpoint=http://execution_client:8551
      --execution-jwt=/secrets/jwt.hex
      --checkpoint-sync-url=https://mainnet.checkpoint.sigp.io
      --disable-deposit-contract-sync

Chain/Network:

Ethereum main network

The RPC call "eth_blockByNumber" returns null for all blocks between 19500000 and 19599999 (included). Moreover, the query works perfectly fine for blocks below 19500000 and greater than 19600000 (include)

#### Expected behaviour

All calls should return a valid result, such as:

> curl $URL -X POST -H "Content-Type: application/json" --data '{"method":"eth_getBlockByNumber","params":[19600000, false],"id":null,"jsonrpc":"2.0"}'

> {"jsonrpc":"2.0","id":null,"result":{"baseFeePerGas":"0x28bddac31" .... }}

#### Actual behaviour between 19500000 and 19599999 (included) return null.

All blocks between

> curl $URL -X POST -H "Content-Type: application/json" --data '{"method":"eth_getBlockByNumber","params":[19500000, false
],"id":1,"jsonrpc":"2.0"}'

> {"jsonrpc":"2.0","id":1,"result":null}

> curl $URL -X POST -H "Content-Type: application/json" --data '{"method":"eth_getBlockByNumber","params":[19599999, false],"id":null,"jsonrpc":"2.0"}'

> {"jsonrpc":"2.0","id":null,"result":null}

#### Steps to reproduce the behaviour

- Sync node.
- Execute the 3 rpc calls above to reproduce the problem.

#### Extra details

Testing another RPC call 'debug_traceBlockByNumber" on block 19500000 produces a more explicit error message:

**block with hash 52345c9721dcf84f8e659f8bda44b93d4e9b003bbd0693bb1bca601bbde3bb26 not found**

> curl $URL -X POST -H "Content-Type: application/json" --data '{"method":"debug_traceBlockByNumber","params":[19500000, {"tracer": "callTracer"}],"id":1,"jsonrpc":"2.0"}'

> {"jsonrpc":"2.0","id":1,"result":null,"error":{"code":-32000,"message":"invalid arguments; block with hash 52345c9721dcf84f8e659f8bda44b93d4e9b003bbd0693bb1bca601bbde3bb26 not found"}}

**Comment by AskAlexSharov** [comment]
show `integration print_stages`

**Comment by jack695** [comment]
Sorry for the late answer, here it is:

/usr/local/bin $ ./integration print_stages --chain mainnet --datadir /data
INFO[05-15|07:59:40.505] logging to file system                   log dir=/data/logs file prefix=integration log level=info json=false
INFO[05-15|07:59:40.507] [db] open                                lable=chaindata sizeLimit=12TB pageSize=8192
INFO[05-15|07:59:44.106] [snapshots:all] Blocks Stat              blocks=19500k indices=19500k alloc=2.3GB sys=2.4GB
INFO[05-15|07:59:44.109] [snapshots:all] Blocks Stat              blocks=0k indices=0k alloc=2.3GB sys=2.4GB
Note: prune_at doesn't mean 'all data before were deleted' - it just mean stage.Prune function were run to this block. Because 1 stage may prune multiple data types to different prune distance.

                                 stage_at        prune_at
Snapshots                        19823919        0
Headers                          19823919        0
BorHeimdall                      0               0
BlockHashes                      19823919        0
Bodies                           19823919        0
Senders                          19823919        0
Execution                        19823919        19823919
Translation                      0               0
HashState                        19823919        0
IntermediateHashes               19823919        19823919
AccountHistoryIndex              19823919        0
StorageHistoryIndex              19823919        0
LogIndex                         19823919        0
CallTraces                       19823919        19823919
TxLookup                         19823919        19465520
Finish                           19823919        0
--
prune distance: 

blocks.v2: true, blocks=19499999, segments=19499999, indices=19499999
blocks.bor.v2: segments=0, indices=0

history.v3: false,  idx steps: 0.00, lastBlockInSnap=0, TxNums_Index(0,1)

sequence: EthTx=2401904366, NonCanonicalTx=0

in db: first header 19600000, last header 19824440, first body 18961979, last body 19824440
--

**Comment by AskAlexSharov** [comment]
@jack695 you can: 
```
integration stage_headers --reset
rm datadir/snapshots/prohibit_new_downloads.lock
```

**Comment by jack695** [comment]
Hey, I don't this worked. After 2 days of re-syncing (Erigon logs since I restarted the node are attached), I still can't fetch blocks starting from 19500000.

One noticeable change is that I can't fetch blocks after 19599999 neither too.


[erigon.log](https://github.com/ledgerwatch/erigon/files/15345034/erigon.log)

`./integration print_stages --datadir /data --chain mainnet
INFO[05-17|07:15:51.823] logging to file system                   log dir=/data/logs file prefix=integration log level=info json=false
INFO[05-17|07:15:51.824] [db] open                                lable=chaindata sizeLimit=12TB pageSize=8192
INFO[05-17|07:15:55.438] [snapshots:all] Blocks Stat              blocks=19500k indices=19500k alloc=2.5GB sys=2.6GB
INFO[05-17|07:15:55.447] [snapshots:all] Blocks Stat              blocks=0k indices=0k alloc=2.5GB sys=2.6GB
Note: prune_at doesn't mean 'all data before were deleted' - it just mean stage.Prune function were run to this block. Because 1 stage may prune multiple data types to different prune distance.

                                 stage_at        prune_at
Snapshots                        19888143        0
Headers                          19888144        0
BorHeimdall                      0               0
BlockHashes                      19888144        0
Bodies                           19888144        0
Senders                          19888144        0
Execution                        19888144        19888144
Translation                      0               0
HashState                        19888144        0
IntermediateHashes               19888144        19888144
AccountHistoryIndex              19888144        0
StorageHistoryIndex              19888144        0
LogIndex                         19888144        0
CallTraces                       19888144        19888144
TxLookup                         19888144        19497690
Finish                           19888144        0
--
prune distance: 

blocks.v2: true, blocks=19499999, segments=19499999, indices=19499999
blocks.bor.v2: segments=0, indices=0

history.v3: false,  idx steps: 0.00, lastBlockInSnap=0, TxNums_Index(0,0)

sequence: EthTx=2392633614, NonCanonicalTx=0

in db: first header 19610000, last header 19888144, first body 19610000, last body 19888144`

**Comment by jack695** [comment]
Hey, any update on this? Is there any fix that would not require to fully resync my node?

**Comment by jack695** [comment]
@Giulio2002 Does your PR (https://github.com/ledgerwatch/erigon/pull/10850) fixes this issue as well as it does for https://github.com/ledgerwatch/erigon/issues/10572?

**Comment by Giulio2002** [comment]
> @Giulio2002 Does your PR (#10850) fixes this issue as well as it does for #10572?

Yes, but you need to run with `--internalcl` for now for a bit, in order for the recovery to happen.
